src/app/frontend/deploy/deployfromsettings_controller.js

```diff
@@ -308,6 +315,14 @@ export default class DeployFromSettingsController {
   isPortMappingFilled_(portMapping) { return !!portMapping.port && !!portMapping.targetPort; }
 
   /**
+   * Returns true when the given environment variable is filled by the user, i.e., is not empty.
+   * @param {!backendApi.EnvironmentVariable} variable
+   * @return {boolean}
+   * @private
+   */
+  isVariableFilled_(variable) { return !!variable.name; }
```

Comment:
Using `!!` in checks could be tricky. It transforms `0` to `false` and it is potential issue. Here it's not a problem since validation doesn't allow non-C identifiers, but for example in port mapping case when one of the ports is `0` it is the problem, because new field doesn't show and the user doesn't know why. We should change it for port mappings I think :)